storagemarket/deal_execution.go

```diff
@@ -51,6 +54,15 @@ func (p *Provider) doDeal(deal *types.ProviderDealState) {
 		bus:      bus,
 	})
 
+	// build in-memory state
+	fi, err := os.Stat(deal.InboundFilePath)
+	if err != nil {
+		err := fmt.Errorf("failed to stat output file: %w", err)
+		p.failDeal(pub, deal, err)
+		return
+	}
+	deal.NBytesReceived = fi.Size()
```

Comment:
it would be great to stream the inbound car through the car-v2 library to build up an index as we're already stream it. that could mean you end up with the file on disk at the end bigger though. can we have this in the interface rather than the file-on-disk?

Reply:
@willscott If the user already has a CARv2 file, just streaming the file over HTTP should get us the Index too, right ? Ah, but you are saying that the Index is never part of the deal size.

Reply:
right. it's probably safer if the storage provider re-validates/generates the index as it pulls it, but the commp will need to be over the carv1 subset, so we need to account for that balance somehow